generated/TOOLS/osxbundle/mpv.app/Contents/Resources/meson.build

```diff
@@ -0,0 +1,8 @@
+input = join_paths(source_root, 'TOOLS', 'osxbundle',
+                   'mpv.app', 'Contents', 'Resources', 'icon.icns')
+osxbundle = custom_target('osxbundle',
+    input: input,
+    output: 'icon.icns.inc',
+    command: [file2string, '@INPUT@', '@OUTPUT@'],
+)
+sources += osxbundle
```

Comment:
this file probably has to be there, but it's kinda weird to have it there. this is the skeleton that is used to create a bundle. so every time a bundle is created we will 'distribute' a build related file with it.

Reply:
I don't think this is the case. The build file here is in the source directory, while meson will (only allow you to) generate the bundle related files in the build directory.

So there will be two directories with this skeleton directory layout.

Reply:
sorry my bad. i totally missed the "generated/". selective reading is great.

Reply:
Well, it could potentially be merged into the meson configuration :) but that's probably a topic to explore after this PR.

Reply:
i think there was a reason why it wasn't part of the waf build either. it's kinda frail and most likely not the best way to create a bundle. so we might not want to add it in the current state at all.

Reply:
Hmm, I guess I don't really know what this is used for. I thought this particular icon was always created for macos.

Reply:
oh yeah, that it is created is right. it's used to set the binary icon if not used in a bundle context, for the "About" dialogue and the "Remote Command Center" integration. since the same icon.

https://github.com/mpv-player/mpv/blob/master/osdep/macosx_application.m#L149-L160
https://github.com/mpv-player/mpv/blob/dd4d239bcb151b08eb81b38486515a3e1222e1f5/video/out/mac/common.swift#L382-L388
https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/osdep/macos/remote_command_center.swift#L108
https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/osdep/macosx_menubar.m#L680

Reply:
I just realized. The build doesn't even install this icon at the moment. Is it meant to do so or is it only ever used as an include?

Reply:
i am not sure what you mean with install? like in a system path in the install step? then no.

the icon is only used in that one included mentioned above or for creating the bundle with this https://github.com/mpv-player/mpv/blob/master/TOOLS/osxbundle.py script.

Reply:
Typically on Linux files would be installed, even icon files to e.g. `/usr/share/icons` as an app icon in the default "hicolor" theme.

On at least windows, it probably gets embedded into GUI applications via the resource compiler, and maybe for macOS too?

(You can do similar things on Linux by using GTK or qt compiled resources and just pointing any "set icon" calls to a resource path instead of a thrme appname. But it's much less common.)

Reply:
then my answer is still the same. no it's not installed. yes it's only used via the generated .h file and its include here https://github.com/mpv-player/mpv/blob/master/osdep/macosx_application.m#L149-L160.

afaik on macOS there isn't anything similar to `/usr/share/icons`, so no installation needed.
the bundle is completely self contained and this is already part of the bundle script mentioned above.

yeah there are similar things with xcode (assets and resource paths) but i believe those are not being embedded in the app binary but in a specific folder hierarchy within the bundle. there are probably other ways to embed them, but none are used with mpv. anyway that was a bit off topic.